client/src/main/java/io/pravega/client/state/impl/StateSynchronizerImpl.java

```diff
@@ -220,16 +220,18 @@ private void conditionallyWrite(Function<StateT, UpdateOrInit<StateT>> generator
             Revision revision = state.getRevision();
             UpdateOrInit<StateT> toWrite = generator.apply(state);
             if (toWrite == null) {
+                log.debug("Nothing to update on segment {}", segment);
                 break;
             }
             Revision newRevision = client.writeConditionally(revision, toWrite);
-            log.trace("Conditionally write returned {} ", newRevision);
+            log.debug("Conditionally write returned {} ", newRevision);
             if (newRevision == null) {
                 fetchUpdates();
             } else {
                 if (!toWrite.isInit()) {
                     applyUpdates(newRevision, toWrite.getUpdates());
                 }
+                log.debug("Updates applied to revision {}", newRevision);
```

Comment:
The updates were no applied to 'newRevision' they were applied to create that revision.

Reply:
Also this is redundant with the above log. Perhaps one of them should be trace. Otherwise they will always appear in pairs logging the same revision information.

Reply:
It will also be logged a third time below.

Reply:
updated the logs to ensure we log only once.